dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -32,143 +23,92 @@ namespace PushNotificationHelpers
+    winrt::event_token PushNotificationChannel::PushReceived(TypedEventHandler<winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel, winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler)
     {
-        if (PushNotificationHelpers::IsPackagedAppScenario())
+        bool registeredEvent { false };
         {
-            if (m_channel)
-            {
-                return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
-                {
-                    if (auto strong = weak_self.get())
-                    {
-                        auto pushArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args);
-                        pushArgs.Handled(true);
-                        handler(*strong, pushArgs);
-                    };
-                });
-            }
-            else
-            {
-                // The channelUri is directly obtained when we request Channel from UDK using RemoteId
-                auto lock = m_lock.lock_exclusive();
-
-                if (!m_foregroundHandlerCount)
-                {
-                    auto appUserModelId{ winrt::Microsoft::Helpers::GetAppUserModelId() };
-
-                    THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), this));
-                }
+            auto lock { m_lock.lock_shared() };
+            registeredEvent = bool(m_foregroundHandlers);
+        }
 
-                ++m_foregroundHandlerCount;
+        if(!registeredEvent)
+        {
+            if(PushNotificationHelpers::IsPackagedAppScenario())
+            {
+                auto appUserModelId{ PushNotificationHelpers::GetAppUserModelId() };
 
-                return m_foregroundHandlers.add(handler);
+                // Register foreground event through the UDK
+                THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), this));
             }
-        }
-        else
-        {
-            auto lock = m_lock.lock_exclusive();
-            if (!m_foregroundHandlerCount++)
+            else
             {
-                auto notificationsLongRunningPlatform{ winrt::Microsoft::Helpers::GetNotificationPlatform() };
+                auto notificationsLongRunningPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
 
                 wil::unique_cotaskmem_string processName;
                 THROW_IF_FAILED(GetCurrentProcessPath(processName));
 
-                THROW_IF_FAILED(notificationsLongRunningPlatform->RegisterForegroundActivator(this, processName.get()));             
+                // Register foreground event through the LRP
+                THROW_IF_FAILED(notificationsLongRunningPlatform->RegisterForegroundActivator(this, processName.get()));
```

Comment:
I would reword this comment:
"Register a Sink with platform brokered through the LongRunningProcess"

Reply:
Avoid use of LRP which doesn't mean much outside of our team.